retrofit/src/main/java/retrofit/Call.java

```diff
@@ -0,0 +1,173 @@
+package retrofit;
+
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.ResponseBody;
+import java.io.IOException;
+import java.net.ProtocolException;
+import java.util.concurrent.Executor;
+
+public final class Call<T> implements Cloneable {
+  private final RestAdapter restAdapter;
+  private final MethodInfo methodInfo;
+  private final Object[] args;
+  private final Executor callbackExecutor; // May be null.
+
+  private volatile com.squareup.okhttp.Call rawCall;
+  private boolean executed; // Guarded by this.
+
+  Call(RestAdapter restAdapter, MethodInfo methodInfo, Object[] args, Executor callbackExecutor) {
+    this.restAdapter = restAdapter;
+    this.methodInfo = methodInfo;
+    this.args = args;
+    this.callbackExecutor = callbackExecutor;
+  }
+
+  Call<T> copyWithExecutor(Executor callbackExecutor) {
+    if (this.callbackExecutor != null) {
+      throw new IllegalStateException("Callback executor already set.");
+    }
+    return new Call<T>(restAdapter, methodInfo, args, callbackExecutor);
+  }
+
+  @SuppressWarnings("CloneDoesntCallSuperClone") // We are a final type & saved from state clearing.
+  @Override public Call clone() {
+    return new Call(restAdapter, methodInfo, args, callbackExecutor);
+  }
+
+  public void enqueue(final Callback<T> callback) {
+    synchronized (this) {
+      if (executed) throw new IllegalStateException("Already executed");
+      executed = true;
+    }
+
+    final com.squareup.okhttp.Call call = createRawCall();
```

Comment:
I see what you're doing here. I think it might actually read better if you deliberately shadow the `rawCall` field? Ie. the local variable would be `rawCall` . . . unsure. 

(I have a hard time reasoning about volatiles, though I think that's just me)